src/cephadm/cephadm

```diff
@@ -2145,10 +2144,14 @@ def create_daemon_dirs(ctx, fsid, daemon_type, daemon_id, uid, gid,
             makedirs(os.path.join(data_dir_root, config_dir, 'data'), uid, gid, 0o755)
 
         # populate the config directory for the component from the config-json
-        for fname in required_files:
-            if 'files' in config_json:  # type: ignore
+        if 'files' in config_json:
+            for fname in config_json['files']:
                 content = dict_get_join(config_json['files'], fname)
-                with open(os.path.join(data_dir_root, config_dir, fname), 'w') as f:
+                if os.path.isabs(fname):
+                    fpath = os.path.join(data_dir_root, fname.lstrip(os.path.sep))
+                else:
+                    fpath = os.path.join(data_dir_root, config_dir, fname)
```

Comment:
I think relying on an absolute path is confusing and differs from how all the other services are implemented.

what about modifying this line to be a relative path (e.g, `alerting/ceph_alerts.yml`) ?
https://github.com/ceph/ceph/blob/381685f17f7fdfc04c2e8a40c09e738054b57fcd/src/pybind/mgr/cephadm/services/monitoring.py#L282